src/main/java/SYSC4806/Controller/HomeController.java

```diff
@@ -40,27 +43,73 @@ public String home(Model model) {
     }
 
     /*
-    * Selects courses based on chosen program and year and returns them to view
-    * to refresh results table
+     * Selects courses based on chosen program, year, courses and catagories and returns them to view
+     * to refresh results table
      */
     @GetMapping("/results/{program}/{year}")
-    public String showResults(Model model, @PathVariable("program") String p, @PathVariable String year, @RequestParam(value = "courses", required = false) String co)
-    {
+    public String showResults(Model model, @PathVariable("program") String p, @PathVariable String year,
+                              @RequestParam(value = "courses", required = false) String co,
+                              @RequestParam(value = "categories", required = false) String ca) {
+
         long p_id = programRepository.findByName(p).getId();
         List<String> courseNames = courseRepository.findCourseByProgramAndYear(p_id, Integer.parseInt(year));
 
-        //Edge case: filter doesnt come up with something
-        if(co!=null){
+        if (co != null && !(co.isEmpty())) {
             List<String> courseNamesFilter = Arrays.asList(co.split("\\s*,\\s*"));
             courseNames.retainAll(courseNamesFilter);
         }
 
+        if (ca != null && !(ca.isEmpty())) {
+            List<String> catNamesFilter = Arrays.asList(ca.split("\\s*,\\s*"));
+
+            //As per the category, finding the list of learning outcomes
+            ArrayList<Long> loListId = new ArrayList<>();
+            for (String cName : catNamesFilter) {
+                Category category = categoryRepository.findByName(cName);
+                loListId.add(learningOutcomeRepository.findByCategoryId(category.getId()).getId());
```

Comment:
I think we should consider reducing the number of queries we have. Is it possible that querying this with just a "findByCategoryName" would be faster? I'm not entirely sure how connection pooling works in spring boot.

But we could change this into like a single SQL query that does:
```sql
SELECT * FROM LearningOutcome lo
LEFT JOIN Category c ON lo.c_id = c.id
WHERE c.name = ?1
```

Where ?1 is a passed parameter of category name and we could just do:
```java
loListId.add(learningOutcomeRepository.findByCategoryName(cName);
```